frontend/app/routes/$lang+/_public+/apply+/$id+/communication-preference.tsx

```diff
@@ -79,49 +89,106 @@ export async function action({ request, params }: ActionFunctionArgs) {
 
   const formSchema = z
     .object({
-      preferredLanguage: z.string({ required_error: 'empty-language' }),
-      preferredMethod: z.string({ required_error: 'empty-method' }),
-      email: z.string().min(1, { message: 'empty-email' }).email({ message: 'invalid-email' }).optional(),
-      confirmEmail: z.string().min(1, { message: 'empty-confirm-email' }).optional(),
-      emailForFuture: z.preprocess((arg) => (arg === '' ? undefined : arg), z.string().email({ message: 'invalid-email' }).optional()),
-      confirmEmailForFuture: z.string().optional(),
+      preferredLanguage: z.enum(['en', 'fr'], { errorMap: () => ({ message: t('apply:communication-preference.error-message.empty-language') }) }),
+      preferredMethod: z.enum(['email', 'mail'], {
+        errorMap: () => ({
+          message: t('apply:communication-preference.error-message.empty-method'),
+        }),
+      }),
```

Comment:
This validation currently works because the values of this `enum(..)` array matches what's in the mock DB for lookups, but these values/IDs can change if/when we plug into the actual API.

I feel for these particular questions where there's only two choices, we would have more control on what we display if we add the option labels in the i18n files and use an `enum` to hold the possible option values, similar to how it's currently done in [`type-of-application.tsx`](https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/routes/%24lang%2B/_public%2B/apply%2B/%24id%2B/type-of-application.tsx)

Reply:
yeah, so I tried to create an enum from an array of values pulled from out lookup service, but TS wasn't happy with it, so I hardcoded the string literals instead.  I'm not super happy with this either.

Reply:
I'll resolve and create a work item for this.